atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -390,6 +391,17 @@ SerializableTransactionManager serializable() {
         return transactionManager;
     }
 
+    private static boolean areTransactionManagerInitializationPrerequisitesSatisfied(
+            AsyncInitializer initializer,
+            LockAndTimestampServices lockAndTimestampServices) {
+        return initializer.isInitialized() && timeLockMigrationCompleteIfNeeded(lockAndTimestampServices);
+    }
+
+    @VisibleForTesting
+    static boolean timeLockMigrationCompleteIfNeeded(LockAndTimestampServices lockAndTimestampServices) {
```

Comment:
Idea to consider: you could avoid the boolean supplier and instead have a `Collection<AsyncInitializer>`, if you passed in `initializer` and `lockAndTimestampServices.migrator()`.

Reply:
Alternatively, you could have all the initialization prerequisite logic in TransactionManagers. Then `isInitialized()` of the `InitializeCheckingWrapper` can simply be `initializationPrerequisite.get()`. Passing in the migrator just to check its status is a bit strange, but both of these seem better than the "some suppliers, some asyncinitializers" approach IMO.

That said, do not feel strongly about this, so I'm also fine with leaving it as it is.